src/target/ch32f1.c

```diff
@@ -152,9 +155,36 @@ static bool ch32f1_flash_lock(target *t)
 	// FLASH_CR_FLOCK_CH32 bit does not exists on *regular* clones and defaults to '0' (see PM0075 for STM32F1xx)
 	if (!(cr & FLASH_CR_FLOCK_CH32)) {
 		DEBUG_WARN("Fast lock failed, cr: 0x%08" PRIx32 "\n", cr);
-		return true;
+		return  false ;
 	}
-	return false;
+	return true;
+}
+/**
+	\fn ch32f1_has_fastUnlock
+	\brief check fastUnlock is there, if so it is a CH32fx
+*/
+static bool ch32f1_has_fastUnlock(target *t)
+{
+	DEBUG_INFO("CH32: has fatunlock \n");
+	// reset fast unlock
+	SET_CR(FLASH_CR_FLOCK_CH32);
+	platform_delay(1); // The flash controller is timing sensitive 
+	uint32_t cr=target_mem_read32(t, FLASH_CR) ;
```

Comment:
Please run clang-format on this code and name the value with more than a 2-letter variable name. This is probably `ctrl` and can + should be made const.